ACPs/103-dynamic-fees/README.md

```diff
@@ -0,0 +1,133 @@
+```text
+ACP: 103
+Title: Add Dynamic Fees to the X-Chain and P-Chain
+Author(s): Dhruba Basu (https://github.com/dhrubabasu), Stephen Buttolph (https//github.com/StephenButtolph), Alberto Benegiamo (https://github.com/abi87)
+Discussions-To: <GitHub Discussion URL (POPULATED BY MAINTAINER, DO NOT SET)>
+Status: Proposed
+Track: Standards
+```
+
+## Abstract
+
+Introduce a dynamic fee mechanism to the X-Chain and the P-Chain. Preview a future transition to a multidimensional fee mechanism.
+
+## Motivation
+
+Under a fixed fee mechanism, there is a greater likelihood of higher network congestion. During periods of high demand, there is no economic incentive for users to withhold issuance of their transactions, extending the period of high demand.
+
+The C-Chain, in [Apricot Phase 3](https://medium.com/avalancheavax/apricot-phase-three-c-chain-dynamic-fees-432d32d67b60), employs a dynamic fee mechanism to raise the price during periods of high demand and lowering the price during periods of low demand. As the price gets too expensive, network utilization will decrease, which drops the price.
+
+The X-Chain and P-Chain currently operate under a fixed fee mechanism. To better account for increased load that may result from [ACP-77](../77-reinventing-subnets/README.md), they should be migrated to a dynamic fee mechanism.
+
+## Specification
+
+### Dimensions
+
+There are four dimensions that will be used to approximate the computational cost ("gas" usage) of a transaction:
+
+1. `Bandwidth` ($B$) is the amount of network bandwidth used in transaction broadcast. This is set to the size of the transaction in bytes.
+2. `Reads` ($R$) is the number of state/database reads used in transaction execution.
+3. `Writes` ($W$) is the number of state/database writes used in transaction execution.
+4. `Compute` ($C$) is the total amount of compute used in transaction execution. This is set to the number of signatures verified during transaction execution.
+
+For each transaction, gas ($G$) can be computed (TODO: The weights for each dimension must be specified prior to this ACP being considered "Implementable"):
+
+$$G = B + R + W + C$$
+
+A future ACP could remove the gas standardization to granularly meter usage of each resource in a multidimensional scheme.
+
+### Mechanism
+
+This mechanism aims to have a target gas usage ($T$) per second and adjusts based on the excess gas usage ($x$), defined as the difference between the current gas usage and $T$.
+
+At the start of building/executing block $b$, $x$ is updated:
+
+$$x = \max(x - (T \cdot \Delta t), 0)$$
+
+Where $\Delta t$ is the number of seconds between $b$ and $b$'s parent block.
+
+The required fee per gas for block $b$ is:
+
+$$M \cdot \exp\left(\frac{x}{K}\right)$$
```

Comment:
can we introduce a notation for "fee rate applied to block b"?
We reuse $b_n$ in a section below with a difference meaning, which does not help readability

Reply:
I generally avoid introducing notation that I don't use. $b_n$ below is not the same as here (and why I didn't use originally).

I'm not sure where the readability concern is, $b_n$ isn't used here and is only defined below

Reply:
We use $b$ here to indicate a block and $b_n$ below to indicate the fee rates associated with the block.
The two notation are too similat to me.
Please introduce the fee rates notation here and reuse it later. Something like
```
fee_rate = $$M \cdot \exp\left(\frac{x}{K}\right)$$
```